tests/Compiler/Validator/Object/AssertDateTimeRangeTest.php

```diff
@@ -0,0 +1,114 @@
+
+        self::assertException(
+            MappingFailedException::class,
+            'Failed to map data at path /: Expected value greater than or equal to 2000-01-05, got 2000-01-04T00:00:00+00:00',
+            static fn() => $validator->map(new DateTimeImmutable('2000-01-04')),
+        );
+    }
+
+    public function testDateTimeRangeValidatorWithExclusiveLowerBound(): void
+    {
+        $validatorCompiler = new AssertDateTimeRange(gt: '2000-01-05');
+        $validator = $this->compileValidator('DateTimeRangeValidatorWithExclusiveLowerBound', $validatorCompiler);
+
+        $validator->map(new DateTimeImmutable('2000-01-06'));
+        $validator->map(null);
+        $validator->map([]);
+
+        self::assertException(
+            MappingFailedException::class,
+            'Failed to map data at path /: Expected value greater than 2000-01-05, got 2000-01-05T00:00:00+00:00',
+            static fn() => $validator->map(new DateTimeImmutable('2000-01-05')),
+        );
+    }
+
+    public function testDateTimeRangeValidatorWithInclusiveUpperBound(): void
+    {
+        $validatorCompiler = new AssertDateTimeRange(lte: '2000-01-05');
+        $validator = $this->compileValidator('DateTimeRangeValidatorWithInclusiveUpperBound', $validatorCompiler);
+
+        $validator->map(new DateTimeImmutable('2000-01-05'));
+        $validator->map(new DateTimeImmutable('2000-01-04'));
+        $validator->map(null);
+        $validator->map([]);
+
+        self::assertException(
+            MappingFailedException::class,
+            'Failed to map data at path /: Expected value less than or equal to 2000-01-05, got 2000-01-06T00:00:00+00:00',
+            static fn() => $validator->map(new DateTimeImmutable('2000-01-06')),
+        );
+    }
+
+    public function testDateTimeRangeValidatorWithExclusiveUpperBound(): void
+    {
+        $validatorCompiler = new AssertDateTimeRange(lt: '2000-01-05');
+        $validator = $this->compileValidator('DateTimeRangeValidatorWithExclusiveUpperBound', $validatorCompiler);
+
+        $validator->map(new DateTimeImmutable('2000-01-04'));
+        $validator->map(null);
+        $validator->map([]);
+
+        self::assertException(
+            MappingFailedException::class,
+            'Failed to map data at path /: Expected value less than 2000-01-05, got 2000-01-05T00:00:00+00:00',
+            static fn() => $validator->map(new DateTimeImmutable('2000-01-05')),
+        );
+    }
+
+    public function testDateTimeRangeValidatorWithInclusiveLowerAndUpperBound(): void
+    {
+        $validatorCompiler = new AssertDateTimeRange(gte: '2000-01-05', lte: '2000-01-10');
+        $validator = $this->compileValidator('DateTimeRangeValidatorWithInclusiveLowerAndUpperBound', $validatorCompiler);
+
+        $validator->map(new DateTimeImmutable('2000-01-05'));
+        $validator->map(new DateTimeImmutable('2000-01-06'));
+        $validator->map(new DateTimeImmutable('2000-01-10'));
```

Comment:
I often check tests to see how something works. Here you use only dates, maybe a usecase with full datetime, different timezones etc could be added.

Reply:
added more tests, better formatting and timezone support